src/manifest.rs

```diff
@@ -17,6 +18,77 @@ pub struct Manifest {
     pub signature: [u8; 64],
 }
 
+/// Parse header and return the version number.
+fn parse_header(header: &[u8]) -> Result<u32> {
+    if header == b"Tako Manifest 1" {
+        Ok(1)
+    } else if header.starts_with(b"Tako Manifest") {
+        let msg = "Manifest version is not supported.";
+        Err(Error::InvalidManifest(msg))
+    } else {
+        let msg = "Manifest does not contain expected 'Tako Manifest 1' header.";
+        Err(Error::InvalidManifest(msg))
+    }
+}
+
+/// Parse a lowercase ascii byte in [0, ..., f], return value as int.
+fn parse_hex(ch: u8) -> Option<u8> {
```

Comment:
Isn't this in Rust's `Data.Char` equivalent somewhere?

Reply:
There is [`u8::from_str_radix()`](https://doc.rust-lang.org/std/primitive.u8.html#method.from_str_radix), but it takes a `str` rather than a single character. It would still work if we slice the string in slices of two characters. Using that to parse hexadecimal byte by byte is about as complicated as doing this I think.